tests/flytekit/unit/core/test_typing_annotation.py

```diff
@@ -0,0 +1,61 @@
+import typing
+from collections import OrderedDict
+
+from flytekit.common.translator import get_serializable
+from flytekit.core import context_manager
+from flytekit.core.annotation import FlyteAnnotation
+from flytekit.core.context_manager import Image, ImageConfig
+from flytekit.core.task import task
+from flytekit.models.annotation import TypeAnnotation
+
+default_img = Image(name="default", fqn="test", tag="tag")
+serialization_settings = context_manager.SerializationSettings(
+    project="project",
+    domain="domain",
+    version="version",
+    env=None,
+    image_config=ImageConfig(default_image=default_img, images=[default_img]),
+)
+entity_mapping = OrderedDict()
+
+
+@task
```

Comment:
can we add some unit tests to `test_type_engine.py` that just does the type conversion without the @task part?  Could we do one of a primitive, one of a typing.List, one of a typing.Dict, and one of a dataclass or something complex?

Reply:
only planning on supporting primitive + List
but sure

Reply:
added tests for primitive + lists in type engine